src/core/public/application/types.ts

```diff
@@ -35,8 +35,65 @@ import { SavedObjectsStart } from '../saved_objects';
 import { AppCategory } from '../../types';
 import { ScopedHistory } from './scoped_history';
 
-/** @public */
-export interface AppBase {
+/**
```

Comment:
Diff can be a little hard to read, but I mostly merged `AppBase` into `App`, as we no longer need this supertype due to the `LegacyApp` removal.